FEC-11079 Additions to ABRSettings API

GouravSna opened the pull request:
### Description of the Changes
- Added filter based on Height, Width and Pixel
- Priority is Height -> Width -> Pixel -> Bitrate (If all the values are passed to ABRSettings)

#### API
Height filter
`player.getSettings().setABRSettings(new ABRSettings().setMinVideoHeight(960).setMaxVideoHeight(540));`
Width filter
`player.getSettings().setABRSettings(new ABRSettings().setMinVideoWidth(960).setMaxVideoWidth(540));`
Pixel based filter
`player.getSettings().setABRSettings(new ABRSettings().setMaxVideoWidth(330).setMaxVideoHeight(190));`
Bitrate filter
`player.getSettings().setABRSettings(new ABRSettings().setMinVideoBitrate(100000).setMaxVideoBitrate(2000000));`

GouravSna committed:
- ABR selection with Height/Width

GouravSna committed:
- ABRSettings with Height and Width

GouravSna committed:
- Removed unused code

GouravSna committed:
Merge branch 'dev' into FEC-11079

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/ABRSettings.java

GouravSna committed:
Merge branch 'dev' into FEC-11079

giladna commented on playkit/src/main/java/com/kaltura/playkit/player/ABRSettings.java:
```diff
@@ -1,61 +1,110 @@
      */
-    private Long maxVideoBitrate = Long.MAX_VALUE;
+    public final static ABRSettings RESET = new ABRSettings()
+            .setInitialBitrateEstimate(0)
```
@GouravSna what is the default in exo?

GouravSna replied:
I have removed it from this API.

GouravSna committed:
- Updated the hardcoded key values.
- Way to reset the initial bitrate estimate.
- Removed initial bitrate from ABRSettings.RESET API

GouravSna committed:
- Javadoc change

GouravSna committed:
- Fixed issue with minABR selection
# Video has 4 tracks
# Set ABR for top 3 tracks
# Update ABR for the least available track
# ABR will show invalid

GouravSna committed:
Merge branch 'dev' into FEC-11079

GouravSna committed:
Merge branch 'dev' into FEC-11079

# Conflicts:
#	playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java

GouravSna committed:
- Handling an edge case, if the bitrate range lies between two tracks then take the min one.

GouravSna committed:
- Optimisation

giladna committed:
Merge branch 'dev' into FEC-11079

GouravSna committed:
- Code optimising for ABR Filtering

GouravSna committed:
- Removed redundant keys

GouravSna committed:
Merge branch 'dev' into FEC-11079

GouravSna committed:
- Code optimising for ABR Filtering

GouravSna committed:
- Switch instead of if/else

GouravSna committed:
- Switch instead of if/else
